linchpin/cli/__init__.py

```diff
@@ -138,7 +139,23 @@ def _write_to_inventory(self, tx_id=None, inv_path=None, inv_format="cfg"):
                         lt_data = targets[name]["inputs"]["layout_data"]
                         t_data = targets[name]["inputs"]["topology_data"]
                         layout = lt_data["inventory_layout"]
-                        i_path = targets[name]["outputs"]["inventory_path"][0]
+                        # check whether inventory_file is mentioned in layout
+                        if layout.get("inventory_file", None):
+                            i_path = layout["inventory_file"]
+                            # if its not absolute path make path
+                            # relative to workspace
+                            if not os.path.isabs(i_path):
+                                # get default variable for inventories_path
+                                d_p = self.ctx.get_evar("default_inventories"
+                                                        "_path")
+                                d_p = Environment().from_string(d_p).render(
+                                    workspace=self.ctx.get_cfg('evars',
+                                                               'workspace'))
+                                i_path = "{0}/{1}".format(d_p, i_path)
```

Comment:
That's a good point in that case we need to have a system in place which renders whole config file with the variables available to us . Since its applicable for the all other config variables. In which case we move away from config parser interpolation to something else. 
We will be needed to device a solution for that .